### PR TITLE
 miscellaneous CMake Fixes

### PR DESCRIPTION
**Pull request by junghans**
## Purpose

Some minor fixes to CMake.

## Author(s)

@junghans

## Backward Compatibility

Yes

## Implementation Notes

N/A

## Post Submission Checklist

- [x] The feature or features in this pull request is complete
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] One or more example input decks are included
- [x] The source code follows the LAMMPS formatting guidelines

## Further Information, Files, and Links

N/A

### CODE REVIEW

**Comment by junghans** [comment]
@stanmoore1 can you have a look at f936ebc2c88c615ced331426ae5af87cd082cf96 adn check if that is fine?

**Comment by stanmoore1** [comment]
@junghans I didn't see this until just now, I'll take a look

**Comment by stanmoore1** [comment]
@junghans I ran the `lib/kokkos/Makefile.kokkos` change through the regression tests and the tests passed.

**Comment by junghans** [comment]
@stanmoore1, thanks! Then let's go ahead with #930 and we merge this later.

**Comment by junghans** [comment]
@sjplimp: This is ready, I will do #920 later.

**Comment by junghans** [comment]
@sjplimp we only need cmake-3.1 for the GPU package, so we can lower the requirement to cmake-2.8.12, which makes build on RHEL7 and CentOs7 easier.

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -129,31 +129,35 @@ foreach(PKG ${ACCEL_PACKAGES} ${OTHER_PACKAGES})
...
+  list(APPEND LAMMPS_LINK_LIBS ${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES})
 endif()
 
-if(PKG_MEAM)
+if(PKG_MEAM OR PKG_USER-H5MD OR PKG_USER-QMMM)
   enable_language(C)
 endif()
 
-if(PKG_KOKKOS OR PKG_MSCG)
+if(PKG_MSCG)
```
Comment (junghans) [inline cmake/CMakeLists.txt]: This block will go away in #920.

**Comment by stanmoore1** [comment]
@junghans is this ready to merge?

**Comment by junghans** [comment]
Yes!

**Comment by stanmoore1** [comment]
OK I've assigned this PR to @sjplimp.

**Pull request merged**